juniper_official/chassis/check-fan-state-rpm.rule

```diff
@@ -0,0 +1,146 @@
+/*
+ * Moniters fan state and RPM percentage and notifies when anomalies are found.
+ * 
+ */
+healthbot {
+    topic chassis {
+        rule check-fan-state-rpm {
+            /*
+             * Moniters fan state and RPM percentage and notifies via dashboard, red color 
+             * when threshold is exceeded else green.
+			 *
+             * Use fan-name as key for rule.
+             */		
+            keys fan-name;
+            synopsis "Fan KPI";
+            description "Checks Fan state and RPM percentage";
+            /*
+             * Sensor configuration to get data from network devices.
+             */			
+            sensor fan-netconf {
+                iAgent {
+                    file fanEVOTable.yml;
+                    table fanEVOTable;
+                    frequency 60s;
+                }
+            }
+            /*
+             * Fields defined using sensor path. Map the longer sensor names
+             * to the shorter field names used in the rules.
+             */			
+            field fan-name {
+                sensor fan-netconf {
+                    path Item;
+                }
+                type string;
+                description "Name of the fan tray and fan number.";
+            }
+            field measurement {
+                sensor fan-netconf {
+                    path measurement;
+                }
+                type integer;
+                description "RPM measurement.";
+            }
+            field rpm-percent {
+                sensor fan-netconf {
+                    path rpm;
+                }
+                type integer;
+                description "RPM percent.";
+            }
+            field status {
+                sensor fan-netconf {
+                    path status;
+                }
+                type string;
+                description "Status of the fan.";
+            }
+            /*
+             * Anomaly detection logic.
+             */			
+            trigger fan-state {
+                frequency 1offset;
+                term state-is-ok {
+                    when {
+                        matches-with "$status" ok {
+                            ignore-case;
+                        }
+                    }
+                    then {
+                        status {
+                            color green;
+                            message "$fan-name is in OK state.";
+                        }
+                    }
+                }
+                term state-is-not-ok {
+                    then {
+                        status {
+                            color red;
+                            message "$fan-name is not in OK state.";
+                        }
+                    }
+                }
+            }
+            trigger rpm-percent {
+                frequency 1offset;
```

Comment:
Trigger synopsis and description missing in most of the rules. please add in all rules